pkg/features/features.go

```diff
@@ -106,7 +107,7 @@ var allFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
 	ManagedIstio:       {Default: true, PreRelease: featuregate.Deprecated},
 	APIServerSNI:       {Default: true, PreRelease: featuregate.Deprecated},
 	SeedChange:         {Default: true, PreRelease: featuregate.Beta},
-	ReversedVPN:        {Default: true, PreRelease: featuregate.Beta},
+	ReversedVPN:        {Default: true, PreRelease: featuregate.Deprecated},
```

Comment:
I think the idea is not to deprecate it but to promote it to GA and to lock it to its default (`true`). Afterwards, in future releases the feature gate can be cleaned up.

```suggestion
	ReversedVPN:        {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```

Reply:
As all code for the legacy VPN is removed with this change, I would consider the feature flag as deprecated.
Making it GA and locking it to a default value seems to me inconsistent.
Making it deprecated results in a warning during gardenlet startup. I would guess, that's something we want to have.
However, locking it to the default, results in gardelet crashing, if someone sets it to false in the gardenlet config.
My feeling is that, it is better to just print a warning and to continue, than crashing in this case.

Reply:
@ialidzhikov thanks for that explanation! I guess, my understanding of feature gates was wrong.
According `LockToDefault`, in which case should it be set to `true`? I see that some GA feature have it set and others don't.

Reply:
As you want to drop the old VPN solution in the same PR, then you need LockToDefault=true. Because ReversedVPN=false wouldn't make sense as the old VPN solution will be removed with this change. Hence, you need LockToDefault=true.
If you just want to promote the ReversedVPN to GA but you still want to preserve the old VPN solution and the possibility the feature gate to be disabled, then you don't need LockToDefault=true.

Reply:
Ah, OK. So we have two kind of GA feature gates: Some that can be changed and some that can't. That contradicts to what I learned about [kubernetes  feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-stages), where GA features can't be disabled.
I guess, there should be a document explaining gardener feature gates, as this was a bit confusing to me.
Anyhow, I'm fine with your suggested change and I'll change it accordingly.